embassy-stm32/src/exti.rs

```diff
@@ -20,9 +44,9 @@ pub struct ExtiPin<T: Instance> {
 }
 
 impl<T: Instance> ExtiPin<T> {
-    pub fn new(mut pin: T, interrupt: T::Interrupt) -> Self {
+    pub fn new(mut pin: T, interrupt: T::Interrupt, syscfg: &mut SysCfg) -> Self {
```

Comment:
We MUST accept `&mut Syscfg` here to ensure that `contrain` has been called. Removing the transmute is simply a convenience.